pkg/controller/release/observe.go

```diff
@@ -77,9 +79,20 @@ func isUpToDate(ctx context.Context, kube client.Client, in *v1beta1.ReleasePara
 	if ocm == nil {
 		return false, errors.New(errChartMetaNilInObservedRelease)
 	}
+
+	in := spec.ForProvider
+
 	if in.Chart.Name != ocm.Name {
 		return false, nil
 	}
+
+	mp := sets.New[xpv1.ManagementAction](spec.ManagementPolicies...)
+
+	if len(mp) != 0 && !mp.HasAny(xpv1.ManagementActionUpdate, xpv1.ManagementActionAll) {
```

Comment:
Should we only do this if feature is enabled?

Reply:
Not sure if that's needed as for example the Kubernetes provider [doesn't check](https://github.com/crossplane-contrib/provider-kubernetes/blob/main/internal/controller/object/object.go#L527) if that feature is enabled.

Reply:
It's probably something we missed there :) 

Not crucial, but it would be good if someone wants to disable this beta feature and revert back to the original flow.

Reply:
Any suggestions how to check if the feature is enabled or disabled inside the `isUpToDate)(` function? How to get to the `controller.Options` there?

Reply:
I have addressed this in the last commit. When I disable the feature (`--no-enable-management-policies`), the `Release` is reporting:

```
NAME            RESOURCE   SYNCED   READY   STATUS
Release/nginx              False    True    ReconcileError: `spec.managementPolicies` is set to a non-default value but the feature is not enabled: [Create Delete Observe]
```

When I remote the `managementPolicies` block, the provider identifies any change to the release and reverts it back.

Reply:
Ahh right, if the management policies are disabled, and there a non-default policy is set, the code wont be executed anyway. There is a safety check in xp-runtime.

And if you have a default management policy, your check here doesnt do anything anyway.

So maybe this check in the provider is not even needed? cc: @turkenh 

We will remove the xp-runtime check when Management Policies go to GA, but then you wont need a provider check anyway as it will be enabled without the possibility to disable.

So IMO you can remove the latest commit, and just double-check if everything works correctly with the Management policies disabled, with using default `managementpolicies`.

Reply:
> So IMO you can remove the latest commit, and just double-check if everything works correctly with the Management policies disabled, with using default managementpolicies.

Ok, let's do so. 

@jtyr I am fine with proceeding if you revert back to your initial implementation, sorry for the trouble 😕

Reply:
@turkenh @lsviben I have reverted the last commit. Please feel free to review it.